mythril/laser/ethereum/gas.py

```diff
@@ -0,0 +1,175 @@
+from ethereum import opcodes
+from ethereum.utils import ceil32
+
+
+def calculate_native_gas(size: int, contract: str):
+    gas_value = None
+    word_num = ceil32(size // 32)
+    if contract == "ecrecover":
+        gas_value = opcodes.GECRECOVER
+    elif contract == "sha256":
+        gas_value = opcodes.GSHA256BASE + word_num * opcodes.GSHA256WORD
+    elif contract == "ripemd160":
+        gas_value = opcodes.GRIPEMD160BASE + word_num * opcodes.GRIPEMD160WORD
+    elif contract == "identity":
+        gas_value = opcodes.GIDENTITYBASE + word_num * opcodes.GIDENTITYWORD
+    else:
+        raise ValueError("Unknown contract type {}".format(contract))
+    return gas_value, gas_value
+
+
+def calculate_sha3_gas(length: int):
+    gas_val = 30 + opcodes.GSHA3WORD * ceil32(length)
```

Comment:
it's gas per word right? shouldn't it be ceil32(length)//32 or something? I am not totally sure

Reply:
Good catch. Fixed it, testing now and then I'll push it with the latest merge :+1:

Reply:
It doesn't seem to be fixed yet @dmuhs

Reply:
Now it is :+1: